### PR TITLE
George 6.0.0

### PR DESCRIPTION
**Pull request by Geokureli**
Simply comparing the two branches for discussion with @EliteMasterEric. I have no intentions of merging this

### CODE REVIEW

File: flixel/FlxCamera.hx [hunk 1]
```diff
@@ -609,6 +609,8 @@ class FlxCamera extends FlxBasic
 			itemToReturn = new FlxDrawItem();
 		}
 
+		if (graphic.isDestroyed) throw 'Attempted to queue an invalid FlxDrawItem, did you destroy a cached sprite?';
+
```
Comment (Geokureli) [inline flixel/FlxCamera.hx:613]: is this triggering for you? My hope with #2974 was to prevent these issues

File: flixel/graphics/FlxGraphic.hx [hunk 2]
```diff
@@ -511,6 +512,8 @@ class FlxGraphic implements IFlxDestroyable
 	 */
 	public inline function getFramesCollections(type:FlxFrameCollectionType):Array<Dynamic>
 	{
+		if (isDestroyed) return [];
```
Comment (Geokureli) [inline flixel/graphics/FlxGraphic.hx:515]: same question, should this have been added in #2974

File: flixel/graphics/tile/FlxDrawQuadsItem.hx [hunk 3]
```diff
@@ -116,6 +116,9 @@ class FlxDrawQuadsItem extends FlxDrawBaseItem<FlxDrawQuadsItem>
 			return;
 
 		var shader = shader != null ? shader : graphics.shader;
+
+		if (shader == null) throw 'Attempted to render an invalid FlxDrawItem, did you destroy a cached sprite?';
+
```
Comment (Geokureli) [inline flixel/graphics/tile/FlxDrawQuadsItem.hx:121]: ditto, re: #2974

File: flixel/system/frontEnds/BitmapFrontEnd.hx [hunk 4]
```diff
@@ -294,7 +294,7 @@ class BitmapFrontEnd
 		if (graphic != null)
 		{
 			removeKey(graphic.key);
-			graphic.destroy();
+			if (!graphic.isDestroyed) graphic.destroy();
```
Comment (Geokureli) [inline flixel/system/frontEnds/BitmapFrontEnd.hx:297]: ditto, re: #2974

File: flixel/text/FlxText.hx [hunk 5]
```diff
@@ -890,7 +890,7 @@ class FlxText extends FlxSprite
 		if (oldWidth != newWidth || oldHeight != newHeight)
 		{
 			// Need to generate a new buffer to store the text graphic
-			var key:String = FlxG.bitmap.getUniqueKey("text");
+			var key:String = FlxG.bitmap.getUniqueKey('text(${this.text})');
```
Comment (Geokureli) [inline flixel/text/FlxText.hx:893]: can you talk about the issue this change solved, also, just made this https://github.com/HaxeFlixel/flixel/issues/3081

File: flixel/graphics/tile/FlxGraphicsShader.hx [hunk 6]
```diff
@@ -62,7 +62,7 @@ class FlxGraphicsShader extends GraphicsShader
 			}
 			return vec4(0.0, 0.0, 0.0, 0.0);
 		}
-	")
+	", true)
```
Comment (Geokureli) [inline flixel/graphics/tile/FlxGraphicsShader.hx:65]: what does this change?

File: flixel/graphics/FlxGraphic.hx [hunk 7]
```diff
@@ -499,7 +499,8 @@ class FlxGraphic implements IFlxDestroyable
 		if (collection.type != null)
 		{
 			var collections:Array<Dynamic> = getFramesCollections(collection.type);
-			collections.push(collection);
+			if (collections.indexOf(collection) == -1)
```
Comment (Geokureli) [inline flixel/graphics/FlxGraphic.hx:502]: Should I add this to flixel? what does it improve?

File: flixel/effects/FlxFlicker.hx [hunk 8]
```diff
@@ -129,6 +129,16 @@ class FlxFlicker implements IFlxDestroyable
...
+	public function pause():Void {
+		if (timer == null) return;
+		timer.active = false;
+	}
+
+	public function resume():Void {
+		if (timer == null) return;
+		timer.active = true;
+	}
+
```
Comment (Geokureli) [inline flixel/effects/FlxFlicker.hx:141]: pretty simple change, why not, same for preventing double release, below
Made an Issue: https://github.com/HaxeFlixel/flixel/issues/3082

**Review by Geokureli — COMMENTED** [review]
comments...

File: flixel/FlxCamera.hx (hunk 1 above)
Comment (Geokureli) [reply flixel/FlxCamera.hx:613]: in discord:
> FlxCamera Line 612 does trigger for me when I try to draw a sprite whose graphic has been disposed

File: flixel/graphics/FlxGraphic.hx (hunk 7 above)
Comment (EliteMasterEric) [reply flixel/graphics/FlxGraphic.hx:502]: I want to say I made this change back when I was working on trying to resolve an issue with FlxAnimate. I'd definitely keep it as you don't want the list of collections to have duplicate references in it (as that can cause the reference counts to get messed up).

**Pull request closed without merging**